servers/rendering/renderer_viewport.cpp

```diff
@@ -118,6 +118,13 @@ void RendererViewport::_configure_3d_render_buffers(Viewport *p_viewport) {
 		} else {
 			float scaling_3d_scale = p_viewport->scaling_3d_scale;
 			RS::ViewportScaling3DMode scaling_3d_mode = p_viewport->scaling_3d_mode;
+
+			if (scaling_3d_scale >= 0.9999 && scaling_3d_scale <= 1.0001) {
```

Comment:
@Calinou I don't know if we omitted this on purpose but we had this check in the past. We always ignored the scaling mode if `scaling_3d_scale` was 1.0. I fear this may have been removed when we added the warning checks below so just re-introducing it. It's very easy to get into a scenario where FSR is applied with a scaling of 1.0 and combined with 2D MSAA this becomes very wasteful.

Reply:
This was actually worse than I thought, it basically removed the existing functionality of disabling scaling which meant that on mobile we never did our most optimal approach anymore.

After discussing with @DarioSamo changed this to `if ((scaling_3d_mode == RS::VIEWPORT_SCALING_3D_MODE_BILINEAR || scaling_3d_mode == RS::VIEWPORT_SCALING_3D_MODE_FSR) && scaling_3d_scale >= 0.9999 && scaling_3d_scale <= 1.0001) {`
For FSR2 and possibly other scaling modes we introduce in the future, there are benefits to using the scaling with a value of 1.0.

This re-instates the old logic when scaling is set to 1.0 but only for the old modes and fixes the backwards compatibility issues.

For the future we should introduce a proper OFF setting that is the default (right now this only exists internally, its not use settable) however properly introducing this would be a major breaking change, so Godot 5?